homeassistant/components/sma/__init__.py

```diff
@@ -1 +1,170 @@
-"""The sma component."""
+"""The sma integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import pysma
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PATH,
+    CONF_SCAN_INTERVAL,
+    CONF_SENSORS,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_CUSTOM,
+    CONF_FACTOR,
+    CONF_GROUP,
+    CONF_KEY,
+    CONF_UNIT,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    PLATFORMS,
+    PYSMA_COORDINATOR,
+    PYSMA_OBJECT,
+    PYSMA_SENSORS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up sma from a config entry."""
+    # Init all default sensors
+    sensor_def = pysma.Sensors()
+
+    if entry.source == "import":
```

Comment:
Would be slightly cleaner to use the `SOURCE_IMPORT` constant, and make this block its own function.

Reply:
Changed to SOURCE_IMPORT and moved the code to two seperate functions. See 055c633